pallets/pips/src/lib.rs

```diff
@@ -803,9 +835,9 @@ decl_module! {
             );
 
             let id = Self::next_pip_id();
-            let pip = PIP {
+            let pip = Pip {
                 id,
-                proposal: *proposal,
+                proposal: *proposal.clone(),
```

Comment:
If we pre-calculate `reportable_proposal_data` and use its argument as ref, we can avoid this clone.